### ISSUE TITLE
The TernaryWeight forward method computes quantized_weight and then applies scaling, but doesn't implement the straight-through estimator (STE) for backpropagation. In PyTorch, the gradient will flow through the quantize_weights operation which contains non-differentiable operations (torch.where). For proper BitNet training, you need to detach the quantized weights and add the original weights back during the backward pass to implement STE correctly.

### ISSUE THREAD

**Issue by tzervas** [tzervas/tritter#13]
The TernaryWeight forward method computes quantized_weight and then applies scaling, but doesn't implement the straight-through estimator (STE) for backpropagation. In PyTorch, the gradient will flow through the quantize_weights operation which contains non-differentiable operations (torch.where). For proper BitNet training, you need to detach the quantized weights and add the original weights back during the backward pass to implement STE correctly.

_Originally posted by @copilot-pull-request-reviewer in https://github.com/tzervas/tritter/pull/2#discussion_r2713634444_
